code/modules/integrated_electronics/subtypes/text.dm

```diff
@@ -0,0 +1,64 @@
+/obj/item/integrated_circuit/text
+	name = "text thingy"
+	desc = "Does text-processing related things."
+	category_text = "Text"
+	complexity = 1
+
+// - Text Replacer - //
+/obj/item/integrated_circuit/text/text_replacer
+	name = "text replacing circuit"
+	desc = "Replaces all of one bit of text with another"
```

Comment:
Give it an input string (haystack), a string to search for (needle), and something to replace the needle with. It will output a modified version of the original string if it finds the needle.